wai-extra/Network/Wai/Middleware/RequestLogger.hs

```diff
@@ -360,6 +372,15 @@ detailedMiddleware' cb DetailedSettings{..} ansiColor ansiMethod ansiStatusCode
               in if null par then [""] else ansiColor White "  Params: " <> par <> ["\n"]
 
   t0 <- getCurrentTime
+
+  -- Optionally prelog the request
+  when mPrelogRequests $ do
+    cb $ mconcat $ map toLogStr $
```

Comment:
Instead of using `mconcat` like this, I'd rather see something that leverages the `Semigroup` instance, which should be more efficient. Something like: `cb $ "PRELOGGING REQUEST: " <> toLogStr (ansiMethod ...) <> " " <> toLogStr (rawPathInfo req) <> ...`